providers/sftp/src/airflow/providers/sftp/sensors/sftp.py

```diff
@@ -89,25 +89,28 @@ def poke(self, context: Context) -> PokeReturnValue | bool:
         if self.file_pattern:
             files_from_pattern = self.hook.get_files_by_pattern(self.path, self.file_pattern)
             if files_from_pattern:
-                actual_files_to_check = [
+                actual_files_present = [
                     os.path.join(self.path, file_from_pattern) for file_from_pattern in files_from_pattern
                 ]
             else:
                 return False
         else:
             try:
-                self.hook.isfile(self.path)
-                actual_files_to_check = [self.path]
-            except OSError as e:
-                if e.errno != SFTP_NO_SUCH_FILE:
-                    raise AirflowException from e
-                actual_files_to_check = []
+                # If a file is present, it is the single element added to the actual_files_present list to be
+                # processed. If the file is a directory, actual_file_present will be assigned an empty list,
+                # since SFTPHook.isfile(...) returns False
+                actual_files_present = [self.path] if self.hook.isfile(self.path) else []
+            except Exception as e:
+                # Previously, this statement the `except` block only handled an OSError. However, the
+                # SFTPHook.isfile handles all OSErrors itself, meaning this block could not be reached.
+                # Instead, a more general exception-handling block has been implemented
```

Comment:
```suggestion
```

Reply:
I don't think we need to explain what used to be here... handling non-`OSError` exceptions is clear enough imo